### PR TITLE
Exporting reports from the web UI

### PR DESCRIPTION
**Pull request by kazet**

### CODE REVIEW

File: artemis/db.py [hunk 1]
```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
             }
         ).replace("\x00", " ")
 
+    def take_single_report_generation_task(self) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            query = session.query(ReportGenerationTask).filter(
+                ReportGenerationTask.status == ReportGenerationTaskStatus.PENDING.value
+            )
+            if query.count() > 0:
```
Comment (bulkowy) [inline artemis/db.py:417]: you can use return `query.first()`, as it gives either first row or None when query returns no row

File: artemis/db.py [hunk 2]
```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
...
+                return query[0]  # type: ignore
+        return None
+
+    def mark_report_generation_task_as_completed(
+        self,
+        task: ReportGenerationTask,
+        status: ReportGenerationTaskStatus,
```
Comment (bulkowy) [inline artemis/db.py:424]: if task status is known from function name, should we not just have `ReportGenerationTaskStatus.done` in function directly? I mean, I dont see reason to pass it here

File: artemis/db.py [hunk 3]
```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
...
+                skip_previously_exported=skip_previously_exported,
+                status=ReportGenerationTaskStatus.PENDING,
+            )
+            session.add(task)
+            session.commit()
+
+    def get_report_generation_task(self, id: int) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            try:
+                return session.query(ReportGenerationTask).get(id)  # type: ignore
```
Comment (bulkowy) [inline artemis/db.py:454]: As above, you are returning None, so for code simplicity you can just pass `.first(<filter>)` which will reduce unnecesary code block

File: artemis/reporting/task_handler.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            report_generation_task.custom_template_arguments,  # type: ignore
+            silent=True,
+        )
+    finally:
+        if previous_reports_directory:
+            shutil.rmtree(previous_reports_directory)
+
+
+def main() -> None:
+
```
Comment (bulkowy) [inline artemis/reporting/task_handler.py:39]:
```suggestion
```

File: artemis/db.py (hunk 2 above)
Comment (kazet) [reply artemis/db.py:424]: changed function name to better reflect its purpose

**Pull request merged**